mesonpy/__init__.py

```diff
@@ -951,6 +952,27 @@ def _project(config_settings: Optional[Dict[Any, Any]]) -> Iterator[Project]:
         yield project
 
 
+def _get_ninja_if_needed() -> List[str]:
+    # only add ninja if it is not present or is too low a version
+    ninja = shutil.which('ninja')
```

Comment:
Note that finding ninja can be slightly complicated. Meson's internal implementation looks like this:

```python
def detect_ninja_command_and_version(version: str = '1.8.2', log: bool = False) -> T.Tuple[T.List[str], str]:
    env_ninja = os.environ.get('NINJA', None)
    for n in [env_ninja] if env_ninja else ['ninja', 'ninja-build', 'samu']:
        prog = ExternalProgram(n, silent=True)
        if not prog.found():
            continue
        try:
            p, found = Popen_safe(prog.command + ['--version'])[0:2]
        except (FileNotFoundError, PermissionError):
            # Doesn't exist in PATH or isn't executable
            continue
[...]
```

Note that on some distros, the command is arbitrarily renamed to "ninja-build" because "ninja" was once owned by an IRC client software last developed in 2002.

And samurai can work as well, under the name `samu`.

Meson will detect either of these cases as fallbacks if "ninja" itself does not exist, or returns a version from an IRC client that is less than 1.8.2.

Reply:
> returns a version from an IRC client that is less than 1.8.2.

Ahh, okay.

Reply:
Probably because "ninja" is the most likely to be found.

Reply:
Yeah, I realized that due to the version check, this order is safe.